From d57decfece425f692735a3ed73454b6762270186 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alexander=20F=C3=A6r=C3=B8y?= <ahf@torproject.org>
Date: Wed, 30 Aug 2023 14:43:21 +0200
Subject: [PATCH] Remove defensive assertion in `relay_key_is_unavailable_()`.

This patch removes a call to `tor_assert_nonfatal_unreached()` in
`relay_key_is_unavailable_()` that is only called when Tor is compiled
without relay support.

Unfortunately, the non-fatal assertion causes a BUG log
message to appear for clients when they start up without relay support
for each CPU worker we spawn. This makes it spotting issues during
bootstrap harder particularly for our iOS developers.

Since the call sites to `get_master_identity_key()` handles `NULL`
values already, we do not think this will be an issue later on.

Reported by Benjamin Erhart (@tla) from Guardian Project.

Fixes tpo/core/tor#40848.
---
 changes/ticket40848            | 4 ++++
 src/feature/relay/routerkeys.h | 1 -
 2 files changed, 4 insertions(+), 1 deletion(-)
 create mode 100644 changes/ticket40848

diff --git a/changes/ticket40848 b/changes/ticket40848
new file mode 100644
index 0000000000..a50a9a028f
--- /dev/null
+++ b/changes/ticket40848
@@ -0,0 +1,4 @@
+  o Minor bugfix (defensive programming):
+    - Disable multiple BUG warnings of a missing relay identity key when
+      starting an instance of Tor compiled without relay support.
+      Fixes bug 40848; bugfix on 0.4.3.1-alpha.
diff --git a/src/feature/relay/routerkeys.h b/src/feature/relay/routerkeys.h
index 7b6d80773c..b97615a9c9 100644
--- a/src/feature/relay/routerkeys.h
+++ b/src/feature/relay/routerkeys.h
@@ -53,7 +53,6 @@ void routerkeys_free_all(void);
 static inline void *
 relay_key_is_unavailable_(void)
 {
-  tor_assert_nonfatal_unreached();
   return NULL;
 }
 #define relay_key_is_unavailable(type) \
-- 
GitLab

